pico/processor/src/main/java/io/helidon/pico/processor/BaseAnnotationProcessor.java

```diff
@@ -0,0 +1,856 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.processor;
+
+import java.io.IOException;
+import java.lang.System.Logger.Level;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.Stack;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.ModuleElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.TypeParameterElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeMirror;
+import javax.tools.JavaFileObject;
+
+import io.helidon.common.Weight;
+import io.helidon.common.types.AnnotationAndValue;
+import io.helidon.common.types.DefaultAnnotationAndValue;
+import io.helidon.common.types.DefaultTypeName;
+import io.helidon.common.types.TypeName;
+import io.helidon.pico.Contract;
+import io.helidon.pico.DefaultServiceInfo;
+import io.helidon.pico.ExternalContracts;
+import io.helidon.pico.InjectionPointInfo;
+import io.helidon.pico.QualifierAndValue;
+import io.helidon.pico.RunLevel;
+import io.helidon.pico.ServiceInfoBasics;
+import io.helidon.pico.tools.AbstractFilerMessager;
+import io.helidon.pico.tools.ActivatorCreatorCodeGen;
+import io.helidon.pico.tools.ActivatorCreatorProvider;
+import io.helidon.pico.tools.ActivatorCreatorRequest;
+import io.helidon.pico.tools.ActivatorCreatorResponse;
+import io.helidon.pico.tools.CodeGenFiler;
+import io.helidon.pico.tools.DefaultActivatorCreator;
+import io.helidon.pico.tools.DefaultActivatorCreatorConfigOptions;
+import io.helidon.pico.tools.DefaultGeneralCreatorRequest;
+import io.helidon.pico.tools.GeneralCreatorRequest;
+import io.helidon.pico.tools.InterceptionPlan;
+import io.helidon.pico.tools.InterceptorCreatorProvider;
+import io.helidon.pico.tools.Messager;
+import io.helidon.pico.tools.Options;
+import io.helidon.pico.tools.ServicesToProcess;
+import io.helidon.pico.tools.ToolsException;
+import io.helidon.pico.tools.TypeNames;
+import io.helidon.pico.tools.TypeTools;
+import io.helidon.pico.tools.spi.ActivatorCreator;
+import io.helidon.pico.tools.spi.InterceptorCreator;
+
+import com.sun.source.util.TreePath;
```

Comment:
We should not be using any `com.sun` packages. Is this part of documented public API that is available on any JDK?

Reply:
https://docs.oracle.com/en/java/javase/16/docs/api/jdk.compiler/com/sun/source/util/TreePath.html